hera_sim/visibilities/healvis_wrapper.py

```diff
@@ -36,64 +38,51 @@ class HealVis(VisibilitySimulator):
     """
 
     point_source_ability = False
+    diffuse_ability = True
+    __version__ = hv.__version__
 
-    def __init__(self, fov=180, nprocesses=1, sky_ref_chan=0, **kwargs):
+    def __init__(self, fov=180, nprocesses=1, sky_ref_chan=0):
         if not HAVE_HEALVIS:
             raise ImportError("to use the healvis wrapper, you must install healvis!")
 
         self.fov = fov
         self._nprocs = nprocesses
         self._sky_ref_chan = sky_ref_chan
 
-        # A bit of a hack here because healvis uses its own AnalyticBeam, and
-        # doesn't check if you are using pyuvsim's one. This should be fixed.
-
-        if "beams" not in kwargs:
-            kwargs["beams"] = [AnalyticBeam("uniform")]
+    def validate(self, model_data: ModelData):
+        """Validate that all data is correct.
 
-        super(HealVis, self).__init__(**kwargs)
+        In addition to standard parameter restrictions, HealVis requires a single beam
+        for all antennae.
+        """
+        if model_data.n_beams > 1:
+            raise ValueError("healvis must use the same beam for all antennas.")
 
         # Check if pyuvsim.analyticbeam and switch to healvis.beam_model
-        if isinstance(self.beams[0], pyuvsim.analyticbeam.AnalyticBeam):
-            old_args = self.beams[0].__dict__
+        # TODO: we shouldn't silently modify model_data.beams...
```

Comment:
Can we add a warning here so we're not silently modifying the beams?

Reply:
Yes, good point. Done.